server/src/main/java/io/crate/execution/dml/TranslogIndexer.java

```diff
@@ -113,9 +139,13 @@ private IndexDocumentBuilder populateLuceneFields(BytesReference source) throws
             Object castValue = valueForInsert(indexer.dataType, entry.getValue());
             if (castValue != null) {
                 indexer.valueIndexer.indexValue(castValue, docBuilder);
-                if (tableIndexSources.containsKey(column)) {
-                    addIndexField(docBuilder, tableIndexSources.get(column), castValue);
+                for (Indexer.IndexColumn indexColumn: indexColumns) {
+                    for (Input<?> input: indexColumn.inputs()) {
+                        TranslogInput translogInput = (TranslogInput) input;
+                        translogInput.prepareValue(castValue, indexer.dataType);
```

Comment:
Wouldn't this add values to indexColumns which aren't related to the column? Also this would override the values for each table column.